src/nvim/eval.c

```diff
@@ -545,6 +546,10 @@ void eval_init(void)
 
   set_vim_var_dict(VV_MSGPACK_TYPES, msgpack_types_dict);
   set_vim_var_dict(VV_COMPLETED_ITEM, dict_alloc());
+
+  dict_T *v_event = dict_alloc();
+  v_event->dv_lock = VAR_FIXED;
```

Comment:
VV_RO on line 381 should be enough, so these two line shouldn't be needed. At least, this works for `v:completed_item`.